eureka-core/src/main/java/com/netflix/eureka/cluster/PeerEurekaNodes.java

```diff
@@ -186,12 +191,15 @@ protected PeerEurekaNode createPeerEurekaNode(String peerEurekaNodeUrl) {
      */
     public static boolean isThisMe(String url) {
         InstanceInfo myInfo = ApplicationInfoManager.getInstance().getInfo();
-        try {
-            URI uri = new URI(url);
-            return (uri.getHost().equals(myInfo.getHostName()));
-        } catch (URISyntaxException e) {
-            logger.error("Error in syntax", e);
-            return false;
+        String hostName = hostFromUrl(url);
```

Comment:
Somewhat paranoid. Are we sure the regex does all the right things w.r.t. the rfcs? Creating a new URI is not so expensive, and we know that it does the right thing underneath.